testament/specs.nim

```diff
@@ -106,7 +106,9 @@ type
     ccodeCheck*: seq[string] ## List of peg patterns that need to be
     ## searched for in the generated code. Used for backend code testing.
     maxCodeSize*: int ## Maximum allowed code size (in bytes) for the test.
-    err*: TResultEnum
+    err*: TResultEnum # REFACTOR Separate into 'expected output' (they can
```

Comment:
This should only be one field on the `TSpec`, which is the expected result. The actual result shouldn't be on the spec at all, that's part of the abuse.

Reply:
Things like `proc codegenCheck(` which accept `given: var TSpec` and then mutate fields will be replaced with report constructors [`88ec8a8` (#486)](https://github.com/nim-works/nimskull/pull/486/commits/88ec8a8a4140abc87b52d6e700504c21c08126f7#diff-a161d4667e86146f2f8003f08f765b8d9580ae92ec5fb6679c80c07a5310a551L1023)